springcloud-streams/src/main/java/org/axonframework/extensions/springcloud/eventhandling/SpringStreamMessageSource.java

```diff
@@ -0,0 +1,150 @@
+     * {@link AxonConfigurationException} if for either of them this holds.
+     *
+     * @param builder the {@link SpringStreamMessageSource.Builder} used to instantiate a {@link SpringStreamMessageSource} instance
+     */
+    protected SpringStreamMessageSource(SpringStreamMessageSource.Builder builder) {
+        builder.validate();
+        if (builder.eventBus != null) {
+            this.messageProcessors.addAll(singletonList(builder.eventBus::publish));
+        }
+        this.converter = builder.converter;
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link SpringStreamMessageSource}.
+     * The {@link SpringMessageConverter} is a <b>hard requirement</b> and thus should be provided.
+     *
+     * @return a Builder to be able to create a {@link SpringStreamMessageSource}.
+     */
+    public static SpringStreamMessageSource.Builder builder() {
+        return new SpringStreamMessageSource.Builder();
+    }
+
+    @Override
+    public Registration subscribe(Consumer<List<? extends EventMessage<?>>> messageProcessor) {
+        messageProcessors.add(messageProcessor);
+        return () -> messageProcessors.remove(messageProcessor);
+    }
+
+    /**
+     * Handles the given {@link Message}. If the filter refuses the message, it is ignored.
+     *
+     * @param message The spring message containing the event to publish
+     */
+    @Override
+    protected void handleMessageInternal(Message<?> message) {
+        Optional<EventMessage<?>> optional = converter.readSpringMessage(message);
+        optional.ifPresent(eventMessage -> {
+            List<? extends EventMessage<?>> messages = singletonList(eventMessage);
+            for (Consumer<List<? extends EventMessage<?>>> messageProcessor : messageProcessors) {
+                messageProcessor.accept(messages);
+            }
+        });
+    }
+
+    /**
+     * Builder class to instantiate a {@link SpringStreamMessageSource}.
+     * The {@link SpringMessageConverter} is a <b>hard requirement</b> and thus should be provided.
+     */
+    public static class Builder {
+
+        private EventBus eventBus;
+        private SpringMessageConverter converter;
+
+        /**
+         * Sets the eventBus for publishing events.
+         *
+         * @param eventBus The messageSource to subscribe
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public SpringStreamMessageSource.Builder eventBus(EventBus eventBus) {
+            this.eventBus = eventBus;
+            return this;
+        }
+
+        /**
+         * Sets the converter to convert spring messages to event messages and versa .
+         *
+         * @param converter The converter to convert messages
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public SpringStreamMessageSource.Builder converter(SpringMessageConverter converter) {
```

Comment:
As we're in the `Builder` here, we do not have to define the root class:

```suggestion
        public Builder converter(SpringMessageConverter converter) {
```

Reply:
@smcvb  No Problem.
Thank you for your reviews.
I'll check your comments and commit again in a few days.
Good luck.

Reply:
Sure thing @mehdichitforoosh! I'll aim to get you an earlier follow-up review this time around :+1: